tests/doc/stringify.ts

```diff
@@ -330,6 +330,36 @@ z:
     expect(String(doc)).toBe('? >+ #comment\n  foo\n\n: bar\n')
   })
 
+  test('Keep pipe when empty block literal is given', () => {
+    const doc = YAML.parseDocument('|-')
+    expect(String(doc)).toBe('|\n\n')
+  })
+
+  test('Keep pipe when pipe is provided', () => {
+    const doc = YAML.parseDocument('|')
+    expect(String(doc)).toBe('|\n\n')
+  })
+
+  test('Keep pipe when empty block literal is given yaml format', () => {
+    const src = source`
+      log4j_2: |-
+    `;
+    const expected = source`
+      log4j_2: |
+
+    `;
+    const doc = YAML.parseDocument(src)
+    expect(String(doc)).toBe(expected)
+  })
+
+  test('Keep pipe when pipe is provided in yaml format', () => {
+    const src = source`
+      log4j_2: |
+    `
+    const doc = YAML.parseDocument(src)
+    expect(String(doc)).toBe(src+'\n')
+  })
```

Comment:
"yaml format" doesn't really mean anything in this context, and these don't add any value beyond the first two tests; drop these.